### DISCUSSION TITLE
the egresspolicy work wrong when the matched pod is running on the non-gateway node.

### DISCUSSION THREAD

**Discussion by hudiehule** [spidernet-io/egressgateway#1587] [Q&A]
1、here is my EgressPolicy. it shows the egress traffic must go out through node bsa5242. 
`apiVersion: v1
items:
- apiVersion: egressgateway.spidernet.io/v1beta1
  kind: EgressPolicy
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"egressgateway.spidernet.io/v1beta1","kind":"EgressPolicy","metadata":{"annotations":{},"name":"test","namespace":"default"},"spec":{"appliedTo":{"podSelector":{"matchLabels":{"app":"egress-demo-te"}}},"egressGatewayName":"default"}}
    creationTimestamp: "2024-12-26T06:26:44Z"
    generation: 1
    name: test
    namespace: default
    resourceVersion: "1854057"
    uid: 90788f0d-ad7b-4741-a3c1-b67918c41cd7
  spec:
    appliedTo:
      podSelector:
        matchLabels:
          app: egress-demo-te
    egressGatewayName: default
    egressIP:
      allocatorPolicy: default
      useNodeIP: false
  status:
    eip:
      ipv4: 10.67.52.44
    node: bsa5242
kind: List
metadata:
  resourceVersion: ""
`
2、then i created a pod matched the policy. when it is running on bsa5242, the egress traffic can go out. but when i run this pod on another node, the egress request cannot return properly.

**Comment by ops-india** [comment]
Same issue, any solution you found?
